### DISCUSSION TITLE
Disable File->Save All when there are no buffers to save

### DISCUSSION THREAD

**Discussion by renejsum** [zed-industries/zed#25485] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

A feature I appreciate in VSCode is that File->Save All is greyed out if all buffers are saved. 

### Environment

Zed: v0.160.7 (Zed)
OS: macOS 12.7.6
Memory: 64 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help present your vision of the feature

Coming from VSCode where it looks like this (1)
But in Zed if all buffers are saved, I can still click Save All. 

<img width="270" alt="image" src="https://github.com/user-attachments/assets/98514e93-a99d-44b8-86d7-830585229f54"><img width="225" alt="image" src="https://github.com/user-attachments/assets/f22c74db-a1c1-4510-8a27-d1068fbc25ed">

Which is nice to check before doing a code reload in docker compose, which takes several seconds

**Comment by notpeter** [comment]
We currently do very little updating of MacOS menu items dynamically at runtime.

It appears VSCode disables the following menu items:
- File->Save All (this request)
- Go->Back/Forward
- Go->Last Edit Location (cmd-k, cmd-q)

Of note, VSCode does not disable these:
- File->Save, File->Revert File
- Edit->Undo, Edit->Redo, Edit->Find 
- Go->Next Problem, Go->Previous Problem
- Splits, etc.
